components/deck/useEditDeck.js

```diff
@@ -156,6 +156,20 @@ const useEditDeck = () => {
       })
       .catch(error => dispatch({ type: 'error', error }))
   }
+
+  const copyDeck = () => {
+    dispatch({ type: 'submit_changes' })
+    const deckObject = state.deck
```

Comment:
I think we would want to update the createdAt field

Reply:
Also, should the deck have the same name, or should it be called something like 'Copy of ...' like in google drive?

Reply:
We should definitely have unique names for decks imo, and we can append "copy" to the name of the deck, and if there are multiple copies, then we can name it "copy", "copy 2", "copy 3", etc. What do you think?

Reply:
Good idea!

For copy 1,2,3 we have to do an extra database call, and some extra logic, let’s do that as a separate pr.